frontend/src/hooks/useApprove.ts

```diff
@@ -0,0 +1,19 @@
+import { TokenPreview } from '@/types'
+import { getBasedContract } from '@/utils/getBaseContract'
+import { Address } from 'viem'
+import { useWriteContract } from 'wagmi'
+
+export const useApprove = (spender: Address) => {
```

Comment:
We should wait for the transaction to be finished - similar to other places with `useWaitForTransactionReceipt`